proposals/lambda-method-group-defaults.md

```diff
@@ -145,31 +190,44 @@ only been supported since C# 10, so only code which has been written since then
 ### Grammar and Parser Changes
 This enhancement requires the following changes to the grammar for lambda expressions.
 ```diff
-explicit_anonymous_function_parameter
+ explicit_anonymous_function_parameter_list
++    : explicit_anonymous_function_parameters
++    | explicit_anonymous_function_parameters ',' parameter_array
++    | parameter_array
++    ;
++
++explicit_anonymous_function_parameters
+     : explicit_anonymous_function_parameter
+       (',' explicit_anonymous_function_parameter)*
+     ;
+
+ explicit_anonymous_function_parameter
 -    : anonymous_function_parameter_modifier? type identifier
 +    : anonymous_function_parameter_modifier? type identifier default_argument?
-    ;
+     ;
 ```
 No changes to the grammar are necessary for method groups since this proposal would only change their semantics.
 
 ### Binder Changes
 
 #### Synthesizing New Delegate Types
-As with the behavior for delegates with `ref` or `out` parameters, a new natural type is generated for each lambda or method group defined with any default parameter values.
+As with the behavior for delegates with `ref` or `out` parameters, a new natural type is generated for each lambda or method group defined with any default or `params` parameters.
```

Comment:
We should spell out how we are updating the prior spec on [function types](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md#natural-function-type) and [delegate types](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md#delegate-types)?

For instance, the delegate types should now say something like:

> The delegate type for the anonymous function or method group with parameter types P1, ..., Pn and return type R is:
> - if any parameter or return value is not by value, **or any parameter is optional or `params`,** or there are more than 16 parameters, or any of the parameter types or return are not valid type arguments (say, (int* p) => { }), then the delegate is a synthesized internal anonymous delegate type with signature that matches the anonymous function or method group, and with parameter names arg1, ..., argn or arg if a single parameter;
> ...


The function type section would now say something like:
> A function_type represents a method signature: the parameter types, **default value, and ref kind and `params` modifier,** and return type and ref kind.

The function type section also says:
```
Function_types are used in a few specific contexts only:
- implicit and explicit conversions
- method type inference (§11.6.3) and best common type (§11.6.3.15)
- var initializers
````
So those sections should be reviewed (and updated as needed) for impact of default values and `params` as well.

Reply:
Alright, thanks, I added how the prior proposal should be updated: 9c3828f58a451d8a8e502933034e6f4032a6727e.

> The function type section also says:
> 
> ```
> Function_types are used in a few specific contexts only:
> - implicit and explicit conversions
> - method type inference (§11.6.3) and best common type (§11.6.3.15)
> - var initializers
> ```
> 
> So those sections should be reviewed (and updated as needed) for impact of default values and `params` as well.

I don't follow. Do you mean to update the section that defines the contexts where *function_type*s are used? I think those stay the same.

Reply:
I meant, make sure to read the rest of the [C# 10 lambda spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md?rgh-link-date=2022-10-13T18%3A30%3A26Z#natural-function-type) and make similar edits to other sections that might be affected.